simpeg/utils/code_utils.py

```diff
@@ -782,6 +783,76 @@ def dep_function(*args, **kwargs):
     return dep_function
 
 
+def _deprecated(msg: str, *, category=DeprecationWarning, stacklevel=1):
```

Comment:
I don’t know if this is really saving any work.
All it’s doing is calling `warnings.warn` ahead of the function call. Why not just not call `warnings.warn` directly inside the function?

Reply:
Ah I see, it comes from [pep702](https://peps.python.org/pep-0702/)

If you really want to get deprecated support for items from static type checkers, I’d say it would be better to just conditionally require `typing_extensions` for Python versions less than 3.13

Reply:
I think it's nice to start using `deprecated`, since it'll probably be the default way to deprecate objects in Python. It also comes with the feature that type checkers can detect used of deprecated objects.

I agree with your point. I'll add `type_extensions` as a dependency then so we can conditionally use their `deprecated` decorator.